keps/sig-scheduling/5598-scheduling-cache/README.md

```diff
@@ -0,0 +1,955 @@
+ cache up to date, plugin owners simply have a new function they need to manage within their plugin, which the cache only aggregates.
+
+ We will also provide tests that evaluate different pod configurations against different node configurations and ensure that
+ any time the signatures match the results do as well. This will help us catch issues in the future, in addition to providing
+ testing opportunities in other areas.
+
+ If plugin changes prove to be an issue, we could codify the signature as a new "Scheduling" object that only has a subset
+ of the fields of the pod. Plugins that "opt-in" could only be given access to this reduced scheduling object, and we could then use the entire scheduling object as the signature. This would make it more or less impossible for the signature and plugins to be out of sync, and would
+ naturally surface new dependencies as additions to the scheduling object. However, as we expect plugin changes to be relatively 
+ modest, we don't believe the complexity of making the interface changes is worth the risk today.
+
+See https://github.com/kubernetes/kubernetes/pull/65714#issuecomment-410016382 as starting point on eCache.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+#### Plugins need to keep signatures up to date
+
+The cache will only work if plugin maintainers are able to keep their portion of the signature up-to-date. We beleive this should
+be doable because the logic is put into the plugin interface itself, and we are restricting it to portions of the pod spec,
+but there is still risk of subtle dependencies creeping in.
+
+If plugin changes prove to be an issue, we could codify the signature as a new "Scheduling" object that only has a subset
+of the fields of the pod. Plugins that "opt-in" could only be given access to this reduced scheduling object, and we could then use the entire scheduling object as the signature. This would make it more or less impossible for the signature and plugins to be out of sync, and would
+naturally surface new dependencies as additions to the scheduling object. However, as we expect plugin changes to be relatively 
+modest, we don't believe the complexity of making the interface changes is worth the risk today.
+
+#### We are narrowing the feature set where the cache will work
+
+Because we are explicitly limiting the functionality that this cache will support, we run the risk of designing something
+that will not work for enough customers for it to be useful.
+
+To mitigate this risk we are actively engaging with customers and doing analysis of data available on K8s users to ensure we are still
+capturing a large enough number of user use cases. We also will address this by expanding over time; we expect to have a few interested
+parties up front, but will then evaluate expansions that could onboard more.
+
+## Design Details
+
+### Pod signature v1
```

Comment:
To discuss risks, please list up all the cluster events that would be ignored by this cache mechanism. I mean, for example, if a node's label is changed while a cache is alive, scores from nodeaffinity could be changed, but we'll ignore them while the cache is alive. If a node taint is removed/added, filter results from tainttoleration could be changed, but we'll ignore them. etc